sway/desktop/output.c

```diff
@@ -392,44 +392,37 @@ void output_damage_whole(struct sway_output *output) {
 
 static void damage_surface_iterator(struct sway_output *output,
 		struct wlr_surface *surface, struct wlr_box *_box, float rotation,
-		void *_data) {
-	bool *data = _data;
-	bool whole = *data;
-
+		void *data) {
+	if (!pixman_region32_not_empty(&surface->buffer_damage)) {
+		return;
```

Comment:
Not sure this is correct. We should still send `frame` events to surfaces that registered a frame callback even if there's no damage.

Reply:
(Note: not sending frame callbacks when the view isn't damaged has caused bugs with mpv in the past)